besu/src/main/java/org/hyperledger/besu/controller/ConsensusScheduleBesuControllerBuilder.java

```diff
@@ -135,14 +141,38 @@ protected ProtocolSchedule createProtocolSchedule() {
     return combinedProtocolScheduleFactory.apply(protocolScheduleSpecs, chainId);
   }
 
+  @Override
+  protected ProtocolContext createProtocolContext(
+      final MutableBlockchain blockchain,
+      final WorldStateArchive worldStateArchive,
+      final GenesisState genesisState,
+      final ProtocolSchedule protocolSchedule,
+      final ConsensusContextFactory consensusContextFactory) {
+    return ScheduableProtocolContext.init(
+        blockchain, worldStateArchive, genesisState, protocolSchedule, consensusContextFactory);
+  }
+
   @Override
   protected ConsensusContext createConsensusContext(
       final Blockchain blockchain,
       final WorldStateArchive worldStateArchive,
       final ProtocolSchedule protocolSchedule) {
-    return besuControllerBuilderSchedule
-        .get(0L)
-        .createConsensusContext(blockchain, worldStateArchive, protocolSchedule);
+    final List<ForkSpec<ConsensusContext>> consensusContextSpecs =
+        besuControllerBuilderSchedule.entrySet().stream()
+            .map(
+                e ->
+                    new ForkSpec<>(
+                        e.getKey(),
+                        e.getValue()
+                            .createConsensusContext(
+                                blockchain, worldStateArchive, protocolSchedule)))
+            .sorted(ForkSpec.COMPARATOR)
```

Comment:
ah nice, this will be useful for the mining coordinator too :)

Reply:
Think I can ditch the sorted now based on #3097